src/eventer/eventer_aco_opset.c

```diff
@@ -162,14 +166,47 @@ eventer_aco_mode_asynch_wrapper(eventer_t e, int mask, void *closure, struct tim
     simple_ctx->func_ops(EVENTER_ASYNCH_CLEANUP, simple_ctx->closure);
   }
   else if(mask == EVENTER_ASYNCH_COMPLETE) {
+    bool zero;
     aco_t *co = simple_ctx->co;
     simple_ctx->func_ops(EVENTER_ASYNCH_COMPLETE, simple_ctx->closure);
+    ck_pr_dec_32_zero(&simple_ctx->gate->ref_cnt, &zero);
     free(simple_ctx);
```

Comment:
You free simple_ctx here, but ctx->gate is still allocated, do not not need to free that as well, or does that get handled in eventer_aco_gate_wait()?  If so, you should probably comment that here